rest/src/main/java/at/jku/isse/ecco/rest/RepositoryHandler.java

```diff
@@ -0,0 +1,158 @@
+package at.jku.isse.ecco.rest;
+
+import at.jku.isse.ecco.core.Variant;
+import at.jku.isse.ecco.feature.Feature;
+import at.jku.isse.ecco.feature.FeatureRevision;
+import at.jku.isse.ecco.rest.classes.RestRepository;
+import at.jku.isse.ecco.service.EccoService;
+
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+
+/** Holds {@see EccoService} with one Repository
+ * EccoService gets initialized at first usage of the Repository.
+ * Each Repository has its own EccoService for Multi user support and performance (loading Repository takes some time)
+ */
+public class RepositoryHandler {
+
+    private final int rId;
+    private final String name;
+    private final Path path;
+    private EccoService service;
+
+    public Path getPath() {
+        return path;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public RepositoryHandler(Path path, int rId) {
+        name = path.getFileName().toString();
+        this.rId = rId;
+        this.path = path;
+    }
+
+    public boolean isInitialized () {
+        return service != null;
+    }
+
+    public RestRepository getRepository() {
+        if(!isInitialized()) {
+            service = new EccoService();
+            service.setRepositoryDir(path.resolve(".ecco"));
+            service.setBaseDir(path);
+            service.open();
+        }
+        return new RestRepository(service, rId, name);
+    }
+
+    public void createRepository() {
+        service = new EccoService();
+        service.setRepositoryDir(path.resolve(".ecco"));
+        service.setBaseDir(path);
+        service.init();
+        new RestRepository(service, rId, name);
+    }
+
+    // Commit ----------------------------------------------------------------------------------------------------------
+    public void addCommit (String message, String config, Path commitFolder, String committer) {
+        service.setBaseDir(commitFolder);
+        service.commit(message, config, committer);
+    }
+
+    //checkout
+    public void checkout(String variantId, Path checkoutPath) {
+        getRepository();
+        service.setBaseDir(checkoutPath);
+        service.checkout(service.getRepository().getVariant(variantId).getConfiguration());
+    }
+
+    // Variant ---------------------------------------------------------------------------------------------------------
```

Comment:
1. unnecessary comment
2. Having to add multiple headers means that this class does too much. Please refactor into, e.g., different facades that are called or something like that. This is true for the other header comments as well

Reply:
after staring at these files intensely, it's probably better to merge it for now and then write tests and refactor them in a separate PR or as a separate issue